### PR TITLE
GAWB-2575/2576: featured/certified methods

### PR DESCRIPTION
**Pull request by MattPutnam**
Corresponding -develop PR: https://github.com/broadinstitute/firecloud-develop/pull/897

I decided to leave the json files as separate files so they're as easy as possible to work with.

Also note that I renamed "featured.json" to "featured-workspaces.json" to avoid confusion. We'll need to rename that in prod when we release.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/method_repo/method_repo_table.cljs [hunk 1]
```diff
@@ -40,17 +59,19 @@
...
                                      :backgroundColor (:background-light style/colors)}
-                             :items [{:label "Public Methods"
+                             :items [{:label "My Methods"
+                                      :predicate :mine?}
+                                     {:label "Public Methods"
                                       :predicate :public}
-                                     {:label "My Methods"
-                                      :predicate (fn [method]
-                                                   (or (not (:public method))
-                                                       (contains? (set (:managers method)) (utils/get-user-email))))}]}
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/method_repo/method_repo_table.cljs:53]: is the ordering switch (Public | My vs. My | Public) intentional? I'm guessing this was in a comp.

File: src/cljs/main/broadfcui/page/method_repo/method_repo_table.cljs [hunk 2]
```diff
@@ -59,32 +80,43 @@
...
+                               :column-data
+                               (fn [{:keys [namespace method-id]}]
+                                 (or (contains? certified-namespaces namespace)
+                                     (contains? certified-methods method-id)))
+                               :as-text
+                               (fn [certified?]
+                                 (when certified? "Certified"))
+                               :render
+                               (fn [certified?]
+                                 (when certified? (icons/certified-icon {:style {:marginLeft -5}})))}
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/method_repo/method_repo_table.cljs]: I know you're just following the mockups, but after clicking around ... I can easily see the next request being "how can I get a list of just the certified methods?" A quick and dirty fix could be just setting `:sortable? false` (and giving the column a header)

**Review by davidangb — COMMENTED** [review]
nice and simple!

re: renaming featured.json to featured-workspaces.json ... before release, let's *copy* featured.json to featured-workspaces.json, then after release delete the file we don't want.

Need to talk to comms to set some initial featured methods; we probably don't want to launch with "Featured Methods (0)".

Please consult with Andrew to determine the appropriate set of tests!

File: src/cljs/main/broadfcui/common/notifications.cljs [hunk 3]
```diff
@@ -195,6 +195,6 @@
                       (.forceUpdate this))))))
    :-get-trial-messages
    (fn [{:keys [state]}]
-     (utils/ajax {:url (config/trial-json-url)
+     (utils/ajax {:url (config/google-bucket-url "trial")
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/common/notifications.cljs]: Should these use `utils/get-google-bucket-file`?

**Review by aednichols — APPROVED** [review]
👍

**Comment by zarsky-broad** [comment]
jenkins retest

**Comment by zarsky-broad** [comment]
tests passed locally!

**Pull request merged**